servers/rendering/renderer_rd/renderer_scene_render_rd.cpp

```diff
@@ -429,8 +429,33 @@ void RendererSceneRenderRD::_render_buffers_post_process_and_tonemap(const Rende
 	can_use_effects &= _debug_draw_can_use_effects(debug_draw);
 	bool can_use_storage = _render_buffers_can_be_storage();
 
-	bool use_fsr = fsr && can_use_effects && rb->get_scaling_3d_mode() == RS::VIEWPORT_SCALING_3D_MODE_FSR;
-	bool use_upscaled_texture = rb->has_upscaled_texture() && rb->get_scaling_3d_mode() == RS::VIEWPORT_SCALING_3D_MODE_FSR2;
+	enum {
+		SCALE_NONE,
+		SCALE_FSR,
+		SCALE_MFX,
+		SCALE_TEMPORAL,
+	} scale_type = SCALE_NONE;
```

Comment:
Bit of a nitpick, but I'm not sure these local-scope enums help readability too much even if it's the intention behind them. For following the code below I have to check what the original viewport scaling enums map to in this scope, and the declaration is also repeated elsewhere in the code with different options.

I feel like it'd be preferrable to use the viewport enums and the extra verbosity would be appreciated, as the current method requires more back and forth in following the logic.

In theory it should not be possible for an invalid scaling mode to reach these functions as it's validated on a higher-level method usually that reverts it to "NONE" if the conditions don't apply. See `RendererViewport::_configure_3d_render_buffers` for further details on this.

Reply:
Fix this and simplified the code considerably. I've left some notes of the changes.